yorkie/backend/db/mongo/decoder.go

```diff
@@ -0,0 +1,119 @@
+package mongo
+
+import (
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/yorkie-team/yorkie/pkg/log"
+	"github.com/yorkie-team/yorkie/yorkie/backend/db"
+)
+
+type decoder interface {
+	Decode(val interface{}) error
+}
+
+func (c *Client) decodeClientInfo(
```

Comment:
I'm wondering why all the functions inside the `decoder.go` file are methods of `Client`. It seems that it can exist as an independent function. The same question exists in `encoder.go`.
Is there any reason to declare it as a method of `Client`?

Reply:
I couldn't think of that far. Since there is no dependence on the client state, a pure function is better.